fastestimator/trace/io/csv_logger.py

```diff
@@ -12,66 +12,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-from typing import Optional, List
-
 import os
+from collections import defaultdict
+from typing import List, Optional, Set, Union
+
+import pandas as pd
 
 from fastestimator.trace import Trace
-from fastestimator.util.util import is_number, to_list
 from fastestimator.util import Data
 
 
 class CSVLogger(Trace):
-    """Log monitored quantity in CSV file manner
+    """Log monitored quantity in CSV file
     Args:
         filename: Output filename.
-        monitor_names: List of key names to monitor. The names can be
-            {"mode", "epoch", "train_step", or output names that other traces
-            create}. If None, it will record all. Defaults to None.
-        separator: Seperator for numbers. Defaults to ", ".
-        append: If true, it will write csv file in append mode. Otherwise, it
-            will overwrite the existed file. Defaults to False.
-        mode: Restrict the trace to run only on given modes {'train', 'eval',
-            'test'}. None will always execute. Defaults to 'eval'.
+        monitor_names: List of key names to monitor, if None then all metrics will be recorded.
+        mode: Restrict the trace to run only on given modes. None will always execute.
     """
     def __init__(self,
                  filename: str,
-                 monitor_names: Optional[List[str]] = None,
-                 separator: str = ", ",
-                 append: bool = False,
-                 mode: str = "eval"):
-        self.keys = monitor_names if monitor_names is None else to_list(monitor_names)
-        super().__init__(inputs="*" if self.keys is None else monitor_names, mode=mode)
-        self.separator = separator
+                 monitor_names: Optional[Union[List[str], str]] = None,
+                 mode: Union[str, Set[str]] = ("eval", "test")):
+        super().__init__(inputs="*" if monitor_names is None else monitor_names, mode=mode)
         self.filename = filename
-        self.append = append
+        self.data = None
 
     def on_begin(self, data: Data):
-        self.file = open(self.filename, 'a' if self.append else 'w')
-        self.file_empty = os.stat(self.filename).st_size == 0
+        self.data = defaultdict(list)
 
     def on_epoch_end(self, data: Data):
-        if self.keys is None:
-            self._infer_keys(data)
-        self._make_header()
-        vals = [data.get(key, "") for key in self.keys]
-        vals = [str(val.numpy()) if hasattr(val, "numpy") else str(val) for val in vals]
-        self.file.write("\n" + self.separator.join(vals))
+        self.data["mode"].append(self.system.mode)
```

Comment:
are we sure that we always want to put mode and epoch into the data? what if the user wants some other system variables?